packages/xo-server-backup-reports/src/index.js

```diff
@@ -100,40 +102,113 @@ const isSkippedError = error =>
   error.message === UNHEALTHY_VDI_CHAIN_ERROR ||
   error.message === NO_SUCH_OBJECT_ERROR
 
-const INDENT = '  '
-const createGetTemporalDataMarkdown = formatDate => (
-  start,
-  end,
-  nbIndent = 0
-) => {
-  const indent = INDENT.repeat(nbIndent)
-
-  const markdown = [`${indent}- **Start time**: ${formatDate(start)}`]
+// ===================================================================
+
+const TITLE_BY_STATUS = {
+  failure: n => `## ${n} Failure${n === 1 ? '' : 's'}`,
+  interrupted: n => `## ${n} Interrupted`,
+  skipped: n => `## ${n} Skipped`,
+  success: n => `## ${n} Success${n === 1 ? '' : 'es'}`,
+}
+
+const getTemporalDataMarkdown = (end, start, formatDate) => {
+  const markdown = [`- **Start time**: ${formatDate(start)}`]
   if (end !== undefined) {
-    markdown.push(`${indent}- **End time**: ${formatDate(end)}`)
+    markdown.push(`- **End time**: ${formatDate(end)}`)
     const duration = end - start
     if (duration >= 1) {
-      markdown.push(`${indent}- **Duration**: ${formatDuration(duration)}`)
+      markdown.push(`- **Duration**: ${formatDuration(duration)}`)
     }
   }
   return markdown
 }
 
-const addWarnings = (text, warnings, nbIndent = 0) => {
-  if (warnings === undefined) {
+const getWarningsMarkdown = (warnings = []) =>
+  warnings.map(({ message }) => `- **${ICON_WARNING} ${message}**`)
+
+const getErrorMarkdown = task => {
+  let message
+  if (
+    task.status === 'success' ||
+    (message = task.result?.message ?? task.result?.code) === undefined
+  ) {
     return
   }
 
-  const indent = INDENT.repeat(nbIndent)
-  warnings.forEach(({ message }) => {
-    text.push(`${indent}- **${ICON_WARNING} ${message}**`)
-  })
+  const label = task.status === 'skipped' ? 'Reason' : 'Error'
+  return `- **${label}**: ${message}`
 }
 
+const MARKDOWN_BY_TYPE = {
+  pool(task, { formatDate }) {
+    const { pool, poolMaster = {} } = task.data
+    const name = pool.name_label || poolMaster.name_label || UNKNOWN_ITEM
+
+    return {
+      body: [
+        `- **UUID**: ${pool.uuid}`,
+        ...getTemporalDataMarkdown(task.end, task.start, formatDate),
+        getErrorMarkdown(task),
+      ],
+      title: `[pool] ${name}`,
+    }
+  },
+  xo(task, { formatDate, jobName }) {
+    return {
+      body: [
+        ...getTemporalDataMarkdown(task.end, task.start, formatDate),
+        getErrorMarkdown(task),
+      ],
+      title: `[XO] ${jobName}`,
+    }
+  },
+  async remote(task, { formatDate, xo }) {
+    const id = task.data.id
+    const name = await xo.getRemote(id).then(
+      ({ name }) => name,
+      error => {
+        logger.warn(error)
+        return UNKNOWN_ITEM
+      }
+    )
+    return {
+      body: [
+        `- **ID**: ${id}`,
+        ...getTemporalDataMarkdown(task.end, task.start, formatDate),
+        getErrorMarkdown(task),
+      ],
+      title: `[remote] ${name}`,
+    }
+  },
+}
+
+const getMarkdown = async (task, props) =>
```

Comment:
Why is it async?

Reply:
To  always returns a promise.

Reply:
Not necessary, `await` behave correctly with non-promises.